### DISCUSSION TITLE
Move the data from extentions (like `ReqData<T>`, but not `Clone`)

### DISCUSSION THREAD

**Discussion by RoDmitry** [actix/actix-web#3050] [Ideas]
Implementation of new `ReqDataMove<T>` (like `ReqData<T>`, but not `Clone`): [request_data_move.rs](https://github.com/RoDmitry/actix-web/blob/req_data_move/actix-web/src/request_data_move.rs)

For example, if I pass my data from middleware, and in my request handler I need an owned value. I think this is very much useful, because _Unlike app data, request data is dropped when the request has finished processing_ ([quote source](https://github.com/actix/actix-web/blob/master/actix-web/src/request_data.rs#L16)). So I don't understand why would I need `Rc<T>`, and where else would I need to use my data, if I would have used `ReqData<Rc<T>>`. Maybe give me examples?
Also if you need to mutate it, it's also easier (because it's owned), and you can later re-insert it back into the extensions map, so you will always have exactly one copy of the data. And if you have forgotten to re-insert it, and you use it elsewhere (but where can it be?), you will get a compile time error (I guess).

Can make it look better and send a PR

**Comment by robjtede** [comment]
It's an interesting idea. I'd consider a PR to [`actix-web-lab`](https://github.com/robjtede/actix-web-lab) for this.
